sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -643,13 +644,13 @@ Models::QueryEntitiesPagedResponse TableClient::QueryEntities(
   auto url = m_url;
   if (!options.NextPartitionKey.empty() && !options.NextRowKey.empty())
   {
-    url.AppendPath(m_tableName);
+    url.AppendPath(Azure::Core::Url::Encode(m_tableName));
     url.AppendQueryParameter("NextPartitionKey", options.NextPartitionKey);
```

Comment:
Why does PartitionKey have to be encoded in `GetEntity` and it does NOT have to be encoded in `QueryEntities`?

Another example where it would be helpful to explain why these parameters are treated differently.